pyro/infer/mcmc/hmc.py

```diff
@@ -325,7 +328,13 @@ def sample(self, params):
         return z.copy()
 
     def diagnostics(self):
-        return OrderedDict([
+        diagnostics = OrderedDict([
             ("step size", "{:.2e}".format(self.step_size)),
-            ("acc. rate", "{:.3f}".format(self._accept_cnt / self._t))
+            ("acc. rate", "{:.3f}".format(self._accept_cnt / self._t)),
         ])
+        if self._divergences:
```

Comment:
My suggestion would be to not add divergence to the progress bar, since this will be available as diagnostic information later. Right now, our progress bar becomes garbled if the terminal width is small and this will further make this limit even larger, forcing the users to have to maximize their terminal to get reasonable looking output. If there is some way we can cram this information within the same number of columns, I'm all for it! :)

Reply:
yup, since we have a diagnostics method, there is no need to expose this in progbar. Let me change it together with a few other points when #1941 is ready.